controllers/admin/auth.controller.js

```diff
@@ -0,0 +1,99 @@
+const Joi = require("joi");
+const User = require("../../models/User");
+const bcrypt = require("bcrypt");
+const firebase = require("../../helpers/auth");
+const {
+  successResponse,
+  errorResponse,
+} = require("../../helpers/responseHelper");
+const {
+  getAuth,
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+  sendEmailVerification,
+} = require("firebase/auth");
+
+const register = async (req, res) => {
+  try {
+    let { email, password, firstName, lastName } = req.body;
+
+    const user = await User.scope("withSecretColumns").findOne({
+      where: { email },
+    });
+    if (user) {
```

Comment:
Change to
`if (user) return throw new Error("User already exists with same email");`